File: GMRatings/GMRatings/RankPickerViewController.m
//
//  RankPickerViewController.m
//  GMRatings
//
//  Created by Matthew Frederick Stradling on 5/22/15.
//  Copyright (c) 2015 Matthew Frederick Stradling. All rights reserved.
//

#import "RankPickerViewController.h"

@interface RankPickerViewController ()

@end

@implementation RankPickerViewController
{
  NSArray *_ranks;
  NSUInteger _selectedIndex;
}

- (void)viewDidLoad {
    [super viewDidLoad];
  
  _ranks = @[@"1 Star",
             @"2 Stars",
             @"3 Stars",
             @"4 Stars",
             @"5 Stars"];
  _selectedIndex = [_ranks indexOfObject:self.rank];
    
    // Uncomment the following line to preserve selection between presentations.
    // self.clearsSelectionOnViewWillAppear = NO;
    
    // Uncomment the following line to display an Edit button in the navigation bar for this view controller.
    // self.navigationItem.rightBarButtonItem = self.editButtonItem;
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

#pragma mark - Table view data source

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
    //#warning Potentially incomplete method implementation.
    // Return the number of sections.
    return 1;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    //#warning Incomplete method implementation.
    // Return the number of rows in the section.
    return [_ranks count];
}


- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
  UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"RankCell"];
  cell.textLabel.text = _ranks[indexPath.row];
  
  if (indexPath.row == _selectedIndex) {
    cell.accessoryType = UITableViewCellAccessoryCheckmark;
  } else {
    cell.accessoryType = UITableViewCellAccessoryNone;
  }
  return cell;
}

#pragma mark - Table view delegate
- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
{
  [tableView deselectRowAtIndexPath:indexPath animated:YES];
  
  if (_selectedIndex != NSNotFound) {
    UITableViewCell *cell = [tableView cellForRowAtIndexPath:[NSIndexPath indexPathForRow:_selectedIndex inSection:0]];
    cell.accessoryType = UITableViewCellAccessoryNone;
  }
  _selectedIndex = indexPath.row;
  
  UITableViewCell *cell = [tableView cellForRowAtIndexPath:indexPath];
  cell.accessoryType = UITableViewCellAccessoryCheckmark;
  NSString *rank = _ranks[indexPath.row];
  [self.delegate rankPickerViewController:self didSelectRank:rank];
}

/*
// Override to support conditional editing of the table view.
- (BOOL)tableView:(UITableView *)tableView canEditRowAtIndexPath:(NSIndexPath *)indexPath {
    // Return NO if you do not want the specified item to be editable.
    return YES;
}
*/

/*
// Override to support editing the table view.
- (void)tableView:(UITableView *)tableView commitEditingStyle:(UITableViewCellEditingStyle)editingStyle forRowAtIndexPath:(NSIndexPath *)indexPath {
    if (editingStyle == UITableViewCellEditingStyleDelete) {
        // Delete the row from the data source
        [tableView deleteRowsAtIndexPaths:@[indexPath] withRowAnimation:UITableViewRowAnimationFade];
    } else if (editingStyle == UITableViewCellEditingStyleInsert) {
        // Create a new instance of the appropriate class, insert it into the array, and add a new row to the table view
    }   
}
*/

/*
// Override to support rearranging the table view.
- (void)tableView:(UITableView *)tableView moveRowAtIndexPath:(NSIndexPath *)fromIndexPath toIndexPath:(NSIndexPath *)toIndexPath {
}
*/

/*
// Override to support conditional rearranging of the table view.
- (BOOL)tableView:(UITableView *)tableView canMoveRowAtIndexPath:(NSIndexPath *)indexPath {
    // Return NO if you do not want the item to be re-orderable.
    return YES;
}
*/

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: SwiftRatings/SwiftRatings/RankPickerViewController.swift
//
//  RankPickerViewController.swift
//  SwiftRatings
//
//  Created by Matthew Frederick Stradling on 6/4/15.
//  Copyright (c) 2015 Matthew Frederick Stradling. All rights reserved.
//
 //ADDED BY MFS //ADDED BY MFS //ADDED BY MFS
import UIKit

class RankPickerViewController: UITableViewController {
  var ranks: [String]!
  var selectedRank:String? = nil
  var selectedRankIndex:Int? = nil
  
  override func viewDidLoad() {
    super.viewDidLoad()
    ranks = ["1 Star",
             "2 Stars",
             "3 Stars",
             "4 Stars",
             "5 Stars"]
    
    if let rank = selectedRank {
      selectedRankIndex = find(ranks, rank)!
    }
  }
  
  override func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
    return ranks.count
  }
  
  override func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
    let cell = tableView.dequeueReusableCellWithIdentifier("RankCell", forIndexPath: indexPath) as! UITableViewCell
    cell.textLabel?.text = ranks[indexPath.row]
    
    if indexPath.row == selectedRankIndex {
      cell.accessoryType = .Checkmark
      cell.textLabel?.backgroundColor = UIColor(red: 0.1, green: 0.6, blue: 0.7, alpha: 1.0)
    } else {
      cell.accessoryType = .None
    }
    return cell
  }

  override func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
    tableView.deselectRowAtIndexPath(indexPath, animated: true)
    
    //Other row is selected - need to deselect it
    if let index = selectedRankIndex {
      let cell = tableView.cellForRowAtIndexPath(NSIndexPath(forRow: index, inSection: 0))
      cell?.accessoryType = .None
      //cell?.textLabel?.backgroundColor = UIColor(white: 1.0, alpha: 1.0)
      cell!.textLabel?.backgroundColor = UIColor(red: 0.1, green: 0.6, blue: 0.7, alpha: 1.0)
    }
    
    selectedRankIndex = indexPath.row
    selectedRank = ranks[indexPath.row]
    
    //update checkmark for current row
    let cell = tableView.cellForRowAtIndexPath(indexPath)
    cell!.accessoryType = .Checkmark
    cell!.textLabel?.backgroundColor = UIColor(red: 0.1, green: 0.6, blue: 0.7, alpha: 1.0)
  }
  
  override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
    if segue.identifier == "SaveSelectedRank" {
      if let cell = sender as? UITableViewCell {
        let indexPath = tableView.indexPathForCell(cell)
        selectedRankIndex = indexPath?.row
        if let index = selectedRankIndex {
          selectedRank = ranks[index]
        }
      }
    }
  }
}
